### PR TITLE
improvement on subscriptions and delete user api

### PR DESCRIPTION
**Pull request by sywhb**
- shows api error message
- soft delete user by calling delete account api
- remove duplicate rss subscriptions by user and url; create a unique constraint

### CODE REVIEW

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/resolvers/importers/uploadImportFileResolver.ts [hunk 1]
```diff
@@ -33,14 +33,14 @@ export const uploadImportFileResolver = authorized<
...
->(async (_, { type, contentType }, { claims: { uid }, log }) => {
+>(async (_, { type, contentType }, { uid }) => {
   if (!VALID_CONTENT_TYPES.includes(contentType)) {
     return {
       errorCodes: [UploadImportFileErrorCode.BadRequest],
     }
   }
 
-  const user = await userRepository.findOneBy({ id: uid })
+  const user = await userRepository.findById(uid)
```
Comment (jacksonh) [inline packages/api/src/resolvers/importers/uploadImportFileResolver.ts:43]: do you need the status here too?

File: packages/db/migrations/0137.do.alter_library_item_tsv_update_trigger.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-- Type: DO
```
Comment (jacksonh) [inline packages/db/migrations/0137.do.alter_library_item_tsv_update_trigger.sql:1]: I guess this is in the other PR too

File: packages/db/migrations/0137.do.alter_library_item_tsv_update_trigger.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+-- Name: alter_library_item_tsv_update_trigger
+-- Description: Alter library_item_tsv_update trigger on omnivore.library_item table to add conditions
+
+BEGIN;
+
+DROP TRIGGER IF EXISTS library_item_tsv_update ON omnivore.library_item;
+
+CREATE TRIGGER library_item_tsv_update
+    BEFORE INSERT OR UPDATE OF readable_content, site_name, title, author, description, note, highlight_annotations
```
Comment (jacksonh) [inline packages/db/migrations/0137.do.alter_library_item_tsv_update_trigger.sql:10]: nice, that is very clean looking

**Review by jacksonh — APPROVED** [review]

**Review by jacksonh — APPROVED** [review]

File: packages/api/src/resolvers/importers/uploadImportFileResolver.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/resolvers/importers/uploadImportFileResolver.ts:43]: @jacksonh i have a custom method `findById` in extended `userRepository` which checks `status` too

Comment (jacksonh) [reply packages/api/src/resolvers/importers/uploadImportFileResolver.ts:43]: Awesome, yeah i was looking for something like that but kind of hard to tell in the diff view.

**Review by jacksonh — APPROVED** [review]

**Pull request merged**